src/ParcelRegistry.Projections.Integration/ParcelVersion/ParcelVersionExtensions.cs

```diff
@@ -0,0 +1,103 @@
+﻿namespace ParcelRegistry.Projections.Integration.ParcelVersion
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Be.Vlaanderen.Basisregisters.EventHandling;
+    using Be.Vlaanderen.Basisregisters.GrAr.Provenance;
+    using Be.Vlaanderen.Basisregisters.ProjectionHandling.Connector;
+    using Be.Vlaanderen.Basisregisters.ProjectionHandling.SqlStreamStore;
+    using System.Linq;
+    using Microsoft.EntityFrameworkCore;
+
+    public static class ParcelVersionExtensions
+    {
+        public static async Task<ParcelVersion> CreateNewParcelVersion<T>(
+            this IntegrationContext context,
+            Guid parcelId,
+            Envelope<T> message,
+            Action<ParcelVersion> applyEventInfoOn,
+            CancellationToken ct,
+            bool renewAddresses = true) where T : IHasProvenance, IMessage
+        {
+            var parcelVersion = await context.LatestPosition(parcelId, ct);
+
+            if (parcelVersion is null)
+            {
+                throw DatabaseItemNotFound(parcelId);
+            }
+
+            var provenance = message.Message.Provenance;
+
+            var newParcelVersion = parcelVersion.CloneAndApplyEventInfo(
+                message.Position,
+                provenance.Timestamp,
+                applyEventInfoOn);
+
+            await context
+                .ParcelVersions
+                .AddAsync(newParcelVersion, ct);
+
+            if (renewAddresses)
+            {
+                var parcelVersionAddresses = await LatestParcelAddress(context, parcelId, parcelVersion.Position, ct);
+                foreach (var parcelVersionAddress in parcelVersionAddresses)
+                {
+                    var newParcelAddressVersion = parcelVersionAddress.CloneAndApplyEventInfo(
+                        message.Position,
+                        provenance.Timestamp);
+
+                    await context
+                        .ParcelVersionAddresses
+                        .AddAsync(newParcelAddressVersion, ct);
+                }
+            }
```

Comment:
Every event should also have corresponding address(es) even if it doesn't change. Except when there are no address attached.

Reply:
this code block will always execute, unless with removals of every address, then we don't add new versions of those addresses
`bool renewAddresses = true`